### PR TITLE
Users/kingoftac/fix nvda clickable announcements

### PR DESCRIPTION
**Pull request by KingOfTac**
# Pull Request

## 📖 Description
This PR fixes an issue where NVDA announces slots as clickable if they have an onClick event bound.

### 🎫 Issues
NVDA would announce the named slots in `Button` as clickable if they contained any nodes because the slots had the onClick event bound to them. More info on the NVDA behavior here: https://github.com/nvaccess/nvda/issues/5830

## 👩‍💻 Reviewer Notes
I removed the `disconnectedCallback` from `Button` since all it was doing was removing the event listeners from the slots. I moved `handleClick` to an event binding in the template, but had to change it from private to public. The `handleClick` method is still marked as internal.

## 📑 Test Plan
I added a new story in storybook with an icon slotted into the start slot to show that the "clickable, clickable" announcement no longer happens. To see the old behavior go to this repro and tab through the buttons. https://stackblitz.com/edit/typescript-qp1vkw?file=index.html

### General

- [X] I have included a change request file using `$ yarn change`
- [ ] I have added tests for my changes.
- [X] I have tested my changes.
- [X] I have updated the project documentation to reflect my changes.
- [X] I have read the [CONTRIBUTING](https://github.com/microsoft/fast/blob/master/CONTRIBUTING.md) documentation and followed the [standards](/docs/community/code-of-conduct/#our-standards) for this project.

### Component-specific

- [ ] I have added a new component
- [X] I have modified an existing component
- [ ] I have updated the [definition file](https://github.com/microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#definition)
- [ ] I have updated the [configuration file](https://github.com/microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#configuration)

## ⏭ Next Steps

Need to check to see if other components are binding click or mousedown events to slots and move them if they are being announced improperly.

### CODE REVIEW

File: packages/web-components/fast-foundation/src/button/button.ts [hunk 1]
```diff
@@ -168,34 +168,13 @@ export class FASTButton extends FormAssociatedButton {
...
 
     /**
      * Prevent events to propagate if disabled and has no slotted content wrapped in HTML elements
      * @internal
      */
-    private handleClick = (e: Event) => {
+    public handleClick = (e: Event) => {
         if (this.disabled && this.defaultSlottedContent?.length <= 1) {
             e.stopPropagation();
         }
```
Comment (chrisdholt) [inline packages/web-components/fast-foundation/src/button/button.ts:180]: Is there a reason we need to keep this? If this is disabled it'll be marked as such on the internal button. I'd suggest removing this and removing the click handler on the button.

**Comment by KingOfTac** [comment]
Would we want to backport this to the current version of FF, fast-compoents, and fluent-components?

File: packages/web-components/fast-foundation/src/button/button.ts (hunk 1 above)
Comment (KingOfTac) [reply packages/web-components/fast-foundation/src/button/button.ts]: Done

**Comment by KingOfTac** [comment]
@chrisdholt I think I borked something with my branch while trying to update it with the latest from master. Let me know if I need to close and re-create the PR.

**Comment by nicholasrice** [comment]
@KingOfTac yeah somehow @bheston's commits got merged in here: https://github.com/microsoft/fast/pull/6461/commits

**Comment by KingOfTac** [comment]
Yeah, I think I'll just nuke this one and recreate it.

**Pull request closed without merging**